### PR TITLE
mprocs 0.6.0 (new formula)

### PR DESCRIPTION
**Pull request by pvolok**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/mprocs.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  head "https://github.com/pvolok/mprocs.git", branch: "master"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/mprocs --version")
```
Comment (carlocab) [inline Formula/mprocs.rb:16]: Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Comment (pvolok) [reply Formula/mprocs.rb:16]: Mprocs is a TUI app. Writing a meaningful test would require spawning mprocs and then passing input to stdin with timeouts. I saw another tui app vifm only checking `--version`. Do I have to come up with some kind of tui interaction emulation?

Comment (pvolok) [reply Formula/mprocs.rb]: I added a test that makes mprocs to run `echo hello mprocs` and makes sure it show the output.

File: Formula/mprocs.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args
+  end
+
+  test do
+    require "pty"
+    require "io/console"
```
Comment (carlocab) [inline Formula/mprocs.rb:17]:
```suggestion
```
I think you never actually make use of this?

**Review by carlocab — APPROVED** [review]
Thanks!

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@pvolok, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**